src/main/java/org/radarcns/management/filters/OAuth2TokenRequestPreZuulFilter.java

```diff
@@ -0,0 +1,80 @@
+package org.radarcns.management.filters;
+
+import com.netflix.zuul.ZuulFilter;
+import com.netflix.zuul.context.RequestContext;
+import java.io.UnsupportedEncodingException;
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import org.radarcns.management.config.ManagementPortalProperties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.codec.Base64;
+import org.springframework.stereotype.Component;
+
+/**
+ * A pre-filter for all request sent to Zuul proxy.
+ * This adds the client credentials and scp
+ */
+@Component
+public class OAuth2TokenRequestPreZuulFilter extends ZuulFilter {
+
+    static final String REFRESH_TOKEN_COOKIE = "rft";
+    @Autowired
+    private  ManagementPortalProperties managementPortalProperties;
+
+    Logger logger = LoggerFactory.getLogger(OAuth2TokenRequestPreZuulFilter.class);
+    @Override
+    public Object run() {
+        RequestContext ctx = RequestContext.getCurrentContext();
+        if (ctx.getRequest().getRequestURI().contains("/oauth/token")) {
```

Comment:
You can move this check to `shouldFilter()`, since the method exists for this purpose.